design-toolkit/components/src/components/pagination/types.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2025 Hypergiant Galactic Systems Inc. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import type { HTMLAttributes, RefAttributes } from 'react';
+import type { VariantProps } from 'tailwind-variants';
+import type { PaginationStyles } from './styles';
+
+export type PaginationProps = RefAttributes<HTMLDivElement> &
+  HTMLAttributes<HTMLDivElement> &
+  VariantProps<typeof PaginationStyles> & {
+    onPreviousPage: () => void;
+    isPreviousPageDisabled: boolean;
+    onNextPage: () => void;
+    isNextPageDisabled: boolean;
+    pageCount: number;
+    pageIndex: number;
+    setPageIndex: (pageIndex: number) => void;
```

Comment:
You could reduce all of these callbacks to a single callback with this signature and the name of onChange